Point_set_processing_3/include/CGAL/estimate_lfs.h

```diff
@@ -0,0 +1,725 @@
+  if (sz == 0)
+      return 0.0;
+  else
+      return std::sqrt(sum_squared_distances / sz);
+}
+
+template <typename Monge_form>
+typename Monge_form::FT
+min_curvature_radius(Monge_form &monge_form)
+{
+  // basic geometric types
+  typedef typename Monge_form::FT FT;
+
+  FT max_principal_curvature = monge_form.principal_curvatures(0);
+  FT min_principal_curvature = monge_form.principal_curvatures(1);
+
+  FT r1 = 1.0 / std::abs(max_principal_curvature);
+  FT r2 = 1.0 / std::abs(min_principal_curvature);
+  FT r = (std::min)(r1, r2);
+
+  return r;
+}
+
+template <typename NeighborQuery>
+typename NeighborQuery::FT
+estimate_local_feature_size(const typename NeighborQuery::Point_3& query, ///< point
+                    typename NeighborQuery::Kernel::Vector_3& normal,
+                    const NeighborQuery& neighbor_query, ///< KD-tree
+                    const typename NeighborQuery::Kernel::Sphere_3& bsphere, ///< bounding sphere
+                    unsigned int jet_k, ///< number of neighbors
+                    typename NeighborQuery::FT neighbor_radius,
+                    unsigned int degree_fitting,
+                    unsigned int degree_monge,
+                    typename NeighborQuery::FT epsilon_band,
+                    typename NeighborQuery::FT apex_angle,
+                    std::size_t N_rays
+                    )
+{
+    // basic geometric types
+    typedef typename NeighborQuery::Kernel Kernel;
+    typedef typename Kernel::Point_3  Point;
+    typedef typename Kernel::Vector_3 Vector;
+    typedef typename Kernel::FT FT;
+
+    // types for jet fitting
+    typedef Monge_via_jet_fitting< Kernel > Monge_jet_fitting;
+    typedef typename Monge_jet_fitting::Monge_form Monge_form;
+
+    std::vector<Point> points;
+    // query using as fallback minimum requires nb points for jet fitting (d+1)*(d+2)/2
+    neighbor_query.get_points (query, jet_k, neighbor_radius, std::back_inserter(points),
+                             (degree_fitting + 1) * (degree_fitting + 2) / 2);
+
+    // estimate jet
+    Monge_jet_fitting monge_fit;
+    Monge_form monge_form = monge_fit(points.begin(), points.end(), degree_fitting, degree_monge);
+
+    if (normal * normal == 0.0)
+        normal = monge_form.normal_direction(); // already normalized in monge_form
+    else
+        monge_form.comply_wrt_given_normal(normal); // orient jet with valid given normal
+
+    FT abs_curvature_r = min_curvature_radius(monge_form);
+
+    // estimate shape diameter
+    FT shape_diameter = estimate_shape_diameter(query, normal, neighbor_query, bsphere, epsilon_band, apex_angle, N_rays);
+    FT half_shape_diameter = 0.5 * shape_diameter;
+
+    FT lfs = (std::min)(abs_curvature_r, half_shape_diameter);
+
+    return lfs;
+}
+
+} /* namespace internal */
+/// \endcond
+
+/**
+   \ingroup PkgPointSetProcessing3Algorithms
+
+   Estimates the local feature size (LFS) for the input 3D point cloud. The function only works for 3D point cloud.
```

Comment:
What do you mean with "The function only works for 3D point cloud."?   That it does not work on a polyhedral surface?

Reply:
it does not work for polyhedral surface

Reply:
Well, this is the point set processing package.  So maybe just drop the sentence (here and in other places).